components/micro-gateway-core/src/main/ballerina/src/gateway/throttler/time_batch.bal

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/streams;
+import ballerina/time;
+
+public type TimeBatch object {
+
+    public string attrExpiredTimestamp;
+    public int timeInMilliSeconds = -1;
+    public streams:LinkedList expiredEventQueue;
+    public streams:Scheduler scheduler;
+    public int expiredEventTime = -1;
+    public int startTime = -1;
+    public any[] windowParameters;
+    public function (streams:StreamEvent?[])? nextProcessPointer;
+
+
+
+    public function __init(function(streams:StreamEvent?[])? nextProcessPointer, any[] windowParameters) {
+        self.scheduler = new(function (@tainted streams:StreamEvent?[] e) {
+                        self.process(e);
+        });
+        self.expiredEventQueue = new;
+        self.attrExpiredTimestamp = "expiryTimestamp";
+        self.windowParameters = windowParameters;
+        self.nextProcessPointer = nextProcessPointer;
+        self.initParameters(windowParameters);
+    }
+
+    public function initParameters(any[] windowParameters) {
+        if (windowParameters.length() == 1) {
+            self.timeInMilliSeconds = <int>windowParameters[0];
+        } else if (windowParameters.length() == 2) {
+            self.timeInMilliSeconds = <int>windowParameters[0];
+            self.startTime = <int>windowParameters[1];
+        } else {
+            error err = error("Can not have more than 2 parameters");
```

Comment:
Length can also be 0 in this case. Should we change the error message to capture that?